packages/legend-query-builder/src/stores/QueryBuilderState.ts

```diff
@@ -284,9 +284,9 @@ export abstract class QueryBuilderState implements CommandRegistrar {
     this.milestoningState.updateMilestoningConfiguration();
   }
 
-  changeMapping(val: Mapping): void {
+  changeMapping(val: Mapping, keepQueryContent?: boolean): void {
```

Comment:
we could maybe add a parameter called options? : { keepQueryContent?: boolean}

we should also disable updating milestone config if this flag is set.